File: medium/2433. Find The Original Array of Prefix Xor/2433. Find The Original Array of Prefix Xor.py
from typing import List


class Solution:
    def findArray(self, pref: List[int]) -> List[int]:
        n = len(pref)
        res = [0] * n
        res[0] = pref[0]

        for i in range(1, n):
            res[i] = pref[i - 1] ^ pref[i]

        return res

File: medium/2433. Find The Original Array of Prefix Xor/2433. Find The Original Array of Prefix Xor.ts
function findArray(pref: number[]): number[] {
    const n = pref.length
    const res = new Array<number>(n).fill(0)
    res[0] = pref[0]

    for (let i = 1; i < n; i++) {
        res[i] = pref[i - 1] ^ pref[i]
    }

    return res
};